protocol/xmrmaker/net.go

```diff
@@ -83,31 +92,23 @@ func (b *Instance) HandleInitiateMessage(msg *net.SendKeysMessage) (net.SwapStat
 		return nil, nil, errAmountProvidedTooHigh
 	}
 
-	if err = b.initiate(offer, offerExtra, common.MoneroToPiconero(providedAmount), common.EtherToWei(msg.ProvidedAmount)); err != nil { //nolint:lll
-		return nil, nil, err
-	}
-
-	offerID, err := types.HexToHash(msg.OfferID)
+	providedPicoXMR := common.MoneroToPiconero(providedAmount)
+	providedWei := common.EtherToWei(msg.ProvidedAmount)
+	state, err := b.initiate(offer, offerExtra, providedPicoXMR, providedWei)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	s, has := b.swapStates[offerID]
-	if !has {
-		panic("did not store swap state in Instance map")
-	}
```

Comment:
The removed code above should have been grabbing the lock before reading from `swapStates`.  The call to `b.initiate` above now returns the state, so we don't need to read it again.